src/v/kafka/server/server.cc

```diff
@@ -948,18 +961,32 @@ ss::future<response_ptr> add_partitions_to_txn_handler::handle(
         } else if (!ctx.authorized(
                      security::acl_operation::write,
                      transactional_id{request.data.transactional_id})) {
-            add_partitions_to_txn_response response{
-              request, error_code::transactional_id_authorization_failed};
+            auto ec = !ctx.audit()
+                        ? error_code::broker_not_available
+                        : error_code::transactional_id_authorization_failed;
+            add_partitions_to_txn_response response{request, ec};
             return ctx.respond(std::move(response));
         }
 
+        // TODO(oren): remove? or do we want an audit check after each round
+        // of authZ checks?
```

Comment:
Question for @michael-redpanda or @graphcareful

Reply:
You only need a single `audit` call once all of the authz checks are performed.  So you can keep the final one.

The ones after authz checks in the conditionals are kind of awkward but based on how the logic works, I think they're fine.